tests/unit/objects/test_secure_files.py

```diff
@@ -0,0 +1,112 @@
+        rsps.add(
+            method=responses.GET,
+            url="http://localhost/api/v4/projects/1/secure_files",
+            json=[secure_file_content],
+            content_type="application/json",
+            status=200,
+        )
+        yield rsps
+
+
+@pytest.fixture
+def resp_create_secure_file():
+    with responses.RequestsMock(assert_all_requests_are_fired=False) as rsps:
+        rsps.add(
+            method=responses.POST,
+            url="http://localhost/api/v4/projects/1/secure_files",
+            json=secure_file_content,
+            content_type="application/json",
+            status=200,
+        )
+        yield rsps
+
+
+@pytest.fixture
+def resp_download_secure_file(binary_content):
+    with responses.RequestsMock(assert_all_requests_are_fired=False) as rsps:
+        rsps.add(
+            method=responses.GET,
+            url="http://localhost/api/v4/projects/1/secure_files/1",
+            json=secure_file_content,
+            content_type="application/json",
+            status=200,
+        )
+        rsps.add(
+            method=responses.GET,
+            url="http://localhost/api/v4/projects/1/secure_files/1/download",
+            body=binary_content,
+            content_type="application/octet-stream",
+            status=200,
+        )
+        yield rsps
+
+
+@pytest.fixture
+def resp_remove_secure_file(no_content):
+    with responses.RequestsMock(assert_all_requests_are_fired=False) as rsps:
+        rsps.add(
+            method=responses.DELETE,
+            url="http://localhost/api/v4/projects/1/secure_files/1",
+            json=no_content,
+            content_type="application/json",
+            status=204,
+        )
+        rsps.add(
+            method=responses.GET,
+            url="http://localhost/api/v4/projects/1/secure_files",
+            json=[],
+            content_type="application/json",
+            status=200,
+        )
+        yield rsps
+
+
+def test_list_secure_files(gl, resp_list_secure_files):
+    secure_files = gl.projects.get(1, lazy=True).secure_files.list()
+    assert len(secure_files) == 1
+    assert secure_files[0].id == 1
+    assert secure_files[0].name == "myfile.jks"
+
+
+def test_create_secure_file(gl, resp_create_secure_file):
+    secure_files = gl.projects.get(1, lazy=True).secure_files.create(
+        {"name": "test", "file": "myfile.jks"}
+    )
+    assert secure_files.id == 1
+    assert secure_files.name == "myfile.jks"
+
+
+def test_download_secure_file(gl, binary_content, resp_download_secure_file):
+    secure_file = gl.projects.get(1, lazy=True).secure_files.get(1)
+    secure_content = secure_file.download()
+    assert isinstance(secure_file, SecureFile)
+    assert secure_content == binary_content
+
+
+def test_remove_secure_file(gl, resp_remove_secure_file):
+    gl.projects.get(1, lazy=True).secure_files.delete(1)
+    secure_files = gl.projects.get(1, lazy=True).secure_files.list()
+    assert len(secure_files) == 0
```

Comment:
Since we're mocking the effect entirely, I'm not sure we need the extra list here. We could just remove `assert_all_requests_are_fired=False` from the fixture for DELETE to ensure the request is correctly called, would also keep the fixture smaller.